### ISSUE TITLE
Delete account feature doesn't ask for a password; displays no error message when incorrect username is entered

### ISSUE THREAD

**Issue by srish** [unstructuredstudio/zubhub#439]
**To Reproduce**

1. Visit your profile page https://zubhub.unstructured.studio/profile
2. Click on the three dots on the top right
3. Click "Delete Account" button that pops up
4. You will notice a dialog that asks for username to confirm deletion. Should we not be asking for password instead?
5. Further, if you try to enter a username that's not yours, nothing happens. You don't event get any error message.

**Expected behavior**
* Ideally, we ask for the password when confirming account deletion. 
* In case an incorrect username is entered, an error message should be displayed.
* Also, there are a few spelling errors in the dialog box. How about we change the text in the dialog to "Are you sure you want to delete your account? You will not be able to undo this action later. If you still want to proceed, type your password into the field below and click DELETE." and change the text of the button to "DELETE".

![Screen Shot 2022-04-21 at 5 34 11 PM](https://user-images.githubusercontent.com/288353/164572661-2efe4f8c-db6c-48b7-b6cd-d0100a5a6eeb.png)

**Comment by Deepthi562** [comment]
@srish i understood what you are referring to, we can do this in two ways, 1)cookies and 2)sessions if the username and password match with the data in our database then we can generate cookie . Am i correct? Can we work this way?

**Comment by alicendeh** [comment]
@srish please can i be assigned this task?

**Comment by srish** [comment]
@Deepthi562 As you have been assigned one task already, I am going to assign this one to @alicendeh. Some tip: There should be a prompt that should ask for username and password that should both be verified once more. We should investigate a bit how other platforms are handling the password challenge to the user and mimic a similar workflow.

**Comment by alicendeh** [comment]
Thank you @srish am working on it

**Comment by alicendeh** [comment]
Hello @srish and @tuxology I have made some progress regarding the issue.

Firstly, I did the text modifications on the pop up and changed the text on the button as required

IMAGE BELOW
<img width="1792" alt="Screen Shot 2022-04-30 at 14 10 25" src="https://user-images.githubusercontent.com/58989178/166106950-fd69a726-797d-4d0a-8584-e50ecda705f3.png">

Also, I added the custom error message when the user enters a wrong username

IMAGE BELOW
<img width="1792" alt="Screen Shot 2022-04-30 at 14 12 34" src="https://user-images.githubusercontent.com/58989178/166107022-5cd1cb29-619c-4d24-b5ce-30dc95728233.png">

So am left to add the password confirmation. 
Please have a look and tell me what you think

**Comment by alicendeh** [comment]
@tuxology please I have a question I checked my browsers console to see the data which is available
 and all I can get is the token, the username,avater and others so id like to ask how do I get the users password from the server in other to do a check when the person enters their information to delete their account?

**Comment by tuxology** [comment]
Hey @NdibeRaymond Can you explain what would be the correct way to take here? Is there a way we can take in the password, make a query again to the backend to re-validate the account?

**Comment by NdibeRaymond** [comment]
> @tuxology please I have a question I checked my browsers console to see the data which is available and all I can get is the token, the username,avater and others so id like to ask how do I get the users password from the server in other to do a check when the person enters their information to delete their account?

Hello @alicendeh, django doesn't store password as plaintext (and this is a good thing). Something that can be tried is,
* to submit a request to delete a user along with the password, then we can do a comparison in the background to find out if the password matches or not (passwords are not stored, but the hash of the passwords are, so we can still hash the incoming password then compare it with the stored hash). If the password matches, we go ahead and delete the account and return a success response. If not, we return an "incorrect password" error.

**Comment by skynette** [comment]
@tuxology is this issue still open? if yes can I work on this for a first issue

**Comment by iamhks** [comment]
Hi @skynette If you need something to refer to while working on this, you can refer my PR: https://github.com/unstructuredstudio/zubhub/pull/637 wherein I have verified the credentials before making the edit-profile change. Simply implementing the same would create a bug if User X enters User Y's credentials and it's accepted. So you need to also verify current username === username entered. Also, just in case you're no longer interested in this PR, do let me know!

**Comment by skynette** [comment]
okay I will give you feedback soon

**Comment by sharon-kathambi** [comment]
Hi @srish  @tuxology i would like to work on this.

**Comment by Dumzy006** [comment]
Hello @srish @tuxology i would like to work on this. Can it be assigned to me please?

**Comment by yokwejuste** [comment]
Hi @srish @tuxology about the error handling and enhancement on this project, I wish to work on this. Have some ideas on what happened, it will definitely be of great help for me to learn.

**Comment by srish** [comment]
Feel free to add a PR without the task assignment as that can happen later :)

**Comment by Michelle-okechi** [comment]
@srish @tuxology I designed the "delete account" feature to ask for a username & password before a user can delete their account.
Secondly, I also designed error states for wrong passwords or usernames

In the first image below, a user is required to enter their username and password before account deletion. I centre-aligned the "cancel" and "delete" buttons for easy usability

![Group 132](https://github.com/unstructuredstudio/zubhub/assets/146777024/39a0e54d-9681-4690-b44c-70c9a9a6f46f)

In the second image, I decided to omit the "Enter username" and left only the "Enter password" as an option because users are already logged in with their usernames and it would be appropriate to ask for only their passwords.

![Group 134](https://github.com/unstructuredstudio/zubhub/assets/146777024/96cace26-d731-478f-872e-9201da89b244)

The third & fourth images show the error states for wrong usernames or passwords

![Group 133](https://github.com/unstructuredstudio/zubhub/assets/146777024/dec4ab79-4d6c-407a-a1a5-87fa75305f27)

![Group 135](https://github.com/unstructuredstudio/zubhub/assets/146777024/486e0d22-cdef-4bfa-afd6-cd0ed41d284f)
